modern-make-handbook-en.md

```diff
@@ -0,0 +1,346 @@
+```make
+deploy:
+	ansible-playbook -i inventory/production --tags "deploy" app-server.yml -vvv --become-user=app --extra-vars=extra.txt --vault-password-file="~/.ansible/vault.txt"
+```
+
+
+
+Next time Bob only needs to say `make deploy` and the magic happens!
+
+Pretty cool, huh? Now Bob:
+
+- does not have to spend another minute to remember the syntax
+- will never use a wrong flag while typing the deploy command
+- is not going to freak out next time, when Ansible decides to rename half of their flags
+
+
+
+Let's do the same to the rest of our commands.
+
+Почему-то то что изначально задумывалось простым, например `rails server`, часто обрастает дополнительными изнуряющими подробностями: `bundle exec bin/rails server -p 3001 RAILS_ENV=development`. 
+
+At some point you start adding details to your simple commands and your `rails server` turns into something like `bundle exec bin/rails server -p 3001 RAILS_ENV=development`.
+
+And we already know what can we do here:
+
+```make
+server:
+	bundle exec bin/rails server -p 3001 RAILS_ENV=development
+```
+
+Moving on:
+
+```make
+logs:
+	tail -f log/development.log
+```
+
+You've got the idea. A good thing about it is that you don't have to do this all at once. Whenever you find yourself struggling to remember that long command just add it to the `Makefile`!
+
+<div style="page-break-after: always;"></div>
+## 2. Overcoming Make weirdness
+
+Bob finally made it to his tests and adds a new line to his `Makefile` to run them conveniently:
+
+```make
+test:
+	MINITEST_REPORTER=SpecReporter bundle exec bin/rails test
+```
+
+But once he tries his new command, a weird thing happens:
+
+```
+$: make test
+make: `test' is up to date.
+```
+
+– "After all it looks like your magical `make` is not *that* magical" - he thinks.
+
+
+Let's get a few decades back. The original idea of `make` was to generate files, and back these days `make test` would mean that you want to generate a file (or a folder) with the name `test`.
+
+`make` is clever enough to check whether such file (or folder) already exists to avoid doing extra work. But this is not what Bob meant!
+
+
+
+To overcome this cleverness of `make` Bob needs to add one magical line into our `Makefile`:
+
+`.PHONY: test`
+
+And if there are several such targets, it's enough to enumerate them one by one using space character:
+
+`.PHONY: app test log doc`
+
+
+
+Another surprise comes when you used spaces instead of tabs for indentation:
+
+```
+$: make test
+Makefile:13: *** missing separator.  Stop.
+```
+
+This does not look right. If your text editor can recognize different file formats, it probably already figured out that it should use tabs. Otherwise, you'll need to tweak it a little.
+
+Now, when Bob went down this rabbit hole and learned all the tricks, he can move on.
+
+<div style="page-break-after: always;"></div>
+## 3. Running multiple commands at once
+
+What if we want to run the tests, and if they pass deploy our code? No problemos:
```

Comment:
No problems?

Reply:
I decided to keep the Spanish phrase that is used in Russian version.

Reply:
Ah right.
But shouldn’t it be “No problemas”?

Reply:
Right, either "no problemas", which is a correct Spanish spelling, or "no problemo" for an [American slang version](https://www.urbandictionary.com/define.php?term=No%20Problemo). I'd probably go for the latter 👍